notebooks/03-query/10-functions-window.ipynb

```diff
@@ -0,0 +1,241 @@
+    "# Window functions are currently experimental. Set the enableWindiowing\n",
+    "# context parameter to \"true\" to use them.\n",
+    "req.add_context(\"enableWindowing\", \"true\")\n",
+    "display.sql(req)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c21a4b38-3e86-4209-bf47-393b72fc4c7e",
+   "metadata": {},
+   "source": [
+    "The following query illustrates all of the built in window functions using the same data:\n",
+    "\n",
+    "* ROW_NUMBER which returns the numeric value for the row within the window\n",
+    "* RANK which returns the numeric rank of the row within the window\n",
+    "* DENSE_RANK which returns the rank of the row with no gaps. For example, if two rows tie for 1, the next row is ranked 2\n",
+    "* PERCENT_RANK which returns the percentage rank of the row within the window according to the formula (rank - 1)/(total window rows - 1)\n",
+    "* CUME_DIST which returns the cumulative distribution of the current row within the window calculated as number of window (rows at the same rank or higher than current row) / (total window rows)\n",
+    "* NTILE which divides the number of results as evently as possible into a number of tiles as and returns the value of tile for the row\n",
+    "* LAG which returns the value for the row that preceeds the current row by a given offset\n",
+    "* LEAD which returns the value for the row that follows the current row by a given offset\n",
+    "* FIRST_VALUE which returns the value for the expression for the first row within the window\n",
+    "* LAST_VALUE which returns the value for the expression for the last row within the window\n",
+    "\n",
+    "See [Window functions](https://druid.apache.org/docs/latest/querying/sql-window-functions) for syntax and more detail.\n",
+    "\n",
+    "The query also demonstrates how you can alias a window within the SELECT clause and define it later with the WINDOW keyword."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "694900d0-891f-41bd-9b45-5ae957385244",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "query = \"\"\"\n",
+    "SELECT FLOOR(__time TO DAY) AS event_time,\n",
+    "    channel,\n",
+    "    ABS(delta) AS change,\n",
+    "    ROW_NUMBER() OVER w AS row_no,\n",
+    "    RANK() OVER w AS rank_no,\n",
+    "    DENSE_RANK() OVER w AS dense_rank_no,\n",
+    "    PERCENT_RANK() OVER w AS pct_rank,\n",
+    "    CUME_DIST() OVER w AS cumulative_dist,\n",
+    "    NTILE(4) OVER w AS ntile_val,\n",
+    "    LAG(ABS(delta), 1, 0) OVER w AS lag_val,\n",
+    "    LEAD(ABS(delta), 1, 0) OVER w AS lead_val,\n",
+    "    FIRST_VALUE(ABS(delta)) OVER w AS first_val,\n",
+    "    LAST_VALUE(ABS(delta)) OVER w AS last_val\n",
+    "FROM example_wikipedia_windows\n",
+    "WHERE channel IN ('#kk.wikipedia', '#lt.wikipedia')\n",
+    "GROUP BY channel, ABS(delta), FLOOR(__time TO DAY) \n",
+    "WINDOW w AS (PARTITION BY channel ORDER BY ABS(delta) ASC)\n",
```

Comment:
Would be cool to see some examples of other PARTITION BY and ORDER BY here - again as someone not too familiar with these functions.